src/TestFramework/TestFramework/Attributes/TestMethod/RetryAttribute.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+/// <summary>
+/// This attribute is used to set a retry count on a test method in case of failure.
+/// </summary>
+[AttributeUsage(AttributeTargets.Method, Inherited = false, AllowMultiple = false)]
+public sealed class RetryAttribute : Attribute
```

Comment:
It'd be good to have this attribute non-sealed so we allow extensibility, which means the logic should also be partially here so it can be overrided. I mainly want that so people (or we?) could ship a retry based off Polly.

Reply:
I think we have a precedens for that with data sources that don't use non-sealed class, but rather than that an interface (ITestDataSource) on the attribute, so any implementer does not have to provide the same ctor parameters we provided.